File: AutoCut.Core/Models/OptimizedSheet.cs
namespace AutoCut.Core.Models;

public class OptimizedSheet
{
    public Sheet Sheet { get; }

    public List<OptimizedPanel> Panels { get; }

    public OptimizedSheet(Sheet sheet, List<OptimizedPanel> panels)
    {
        Sheet = sheet;
        Panels = panels;
    }

    public static OptimizedSheet Empty(Sheet sheet) =>
        new OptimizedSheet(sheet, new List<OptimizedPanel>());
}

File: src/core/optimizedSheet.ts
import type { Cut } from "./cut";
import type { OptimizedPanel } from "./optimizedPanel";
import type { Sheet } from "./sheet";
import type { FreeSpace } from "./freeSpace";

export type OptimizedSheet = {
	sheet: Sheet;
	panels: OptimizedPanel[];
	cuts: Cut[];
	freeSpaces: FreeSpace[];
};

// TODO: maybe replace this with a class called sheet and type sheet with sheetTemplate.
// Then this class could have a constructor that takes a sheetTemplate